### PR TITLE
null out currentSource_ on tech dispose and src change

### PR DESCRIPTION
**Pull request by gkatsev**
Follow up to https://github.com/videojs/video.js/pull/3285, we need `currentSource_` to be cleared out if we're switching sources on a tech and we're no longer using a source handler. Plus, when the tech is disposed, we should clear out the `currentSource_`.

### CODE REVIEW

File: src/js/tech/tech.js [hunk 1]
```diff
@@ -813,6 +813,8 @@ Tech.withSourceHandlers = function(_Tech){
     // than clear all of our current tracks
     if (this.currentSource_) {
       this.clearTracks(['audio', 'video']);
```
Comment (gkatsev) [inline src/js/tech/tech.js]: @BrandonOCasey is there a reason why we don't want to just *always* call `clearTracks()` here?

Comment (BrandonOCasey) [inline src/js/tech/tech.js]: I'm not sure I entirely remember but I think it had to do with saving keeping Audio/Video Tracks that were added before we had a currentSource

Comment (gkatsev) [inline src/js/tech/tech.js]: Ok, I'll see what I can find out. Currently, the tests break because of this.

Comment (gkatsev) [inline src/js/tech/tech.js]: Ended up removing clearing out the `currentSource_` in `disposeSourceHandler` and only doing it inside this if statement.

**Comment by incompl** [comment]
Looks good to me

**Comment by misteroneill** [comment]
LGTM

**Pull request closed without merging**